src/lightcurvelynx/graph_state.py

```diff
@@ -502,13 +503,19 @@ def extract_single_sample(self, sample_num):
         # Make a copy of the GraphState with exactly one sample.
         new_state = GraphState(1)
         new_state.num_parameters = self.num_parameters
+        new_state.sample_offset = self.sample_offset
```

Comment:
Is it related to https://github.com/lincc-frameworks/lightcurvelynx/pull/623?

Reply:
Yes. It is technically copying meta-data that was added in #625 (merged today). That PR was laying the ground work for a few different approaches, including #623 (but useful outside that too)